src/sysen2/mldev.108

```diff
@@ -166,22 +194,45 @@ notits:	;; nothing came of it, fall through
 			.SUSET [.RUNAME,,JBCFN2]
 			JRST (A) ]
 		MOVEI A,'ML		;USE THE MATHLAB LINE PRINTER
-		JRST GOTHST ]
+		JRST GOTHSX ]
 ;HOST NOT RECOGNIZED
 NSD:	MOVSI A,%ENSDV
 	JRST NOGO2
 
+hhhddd:				; get device from last 3 characters
+	hrlz c,a		; get device and left justify
+	movem c,jbcdev		; save remote device
+	hllzs a			; get hostname
+	jrst gothst		; process host
+
+doshoe:				; device is SHOE
+	lsh c,12.		; left justfify device
+	movem c,jbcdev		; set remote device
+	ldb a,[301400,,a]	; get 2-character hostname
+	lsh a,30		; left justify hostname
+	jrst gothst		; process host
```

Comment:
Maybe "gothst" should test if it is an ITS name, but I'm not sure.

Reply:
It doesn't really matter.  GOTHST will convert the hostname to ascii and then look it up in the host table. If this fails, then we'll fail the call with a "NO SUCH DEVICE".  As mentioned above, the reason for CHKITS is really to handle the cases of ambiguity.  I find the chkits (and itsnms) a little "hacky" anyway -- since it requires rebuilding ITS in order to add a new known host.  I think it is actually better to simply use a hostname lookup to determine if the host is a known host.  The failure of MLDEV in either case is the same.  The only thing that CHKITS really gives us is some control over ambiguity and a way to "escape" to the "translated" device handling (e.g. TPL, XGP, etc.).